### PR TITLE
Detach Stealth Mode [Spec + Impl]

### PR DESCRIPTION
**Pull request by rhuss**
## Summary

Replace the spex-detach extension's pr/ branch stripping mechanism with .git/info/exclude stealth mode. Spec files are never committed to the code repo, making leaks structurally impossible for upstream/brownfield contributions.

## Changes

- **New `enable` subcommand**: Writes .specify/, specs/, brainstorm/ to .git/info/exclude (idempotent, preserves existing entries)
- **Simplified `archive` subcommand**: Copy-and-commit to sibling specs repo
- **Removed**: `detach`, `verify`, `clean-branch-name` subcommands (pr/ branch stripping)
- **New `after_init` hook**: Auto-enables stealth mode during `specify init`
- **Updated `before_finish` hook**: Archives specs to sibling repo
- **33 new integration tests**
- **Updated docs**: README.md, help.md

## Artifacts

- Spec: `specs/045-detach-stealth-mode/spec.md`
- Plan: `specs/045-detach-stealth-mode/plan.md`
- Tasks: `specs/045-detach-stealth-mode/tasks.md`

Assisted-By: 🤖 Claude Code

## Summary by CodeRabbit

* **New Features**
  * Added stealth mode to hide spec artifacts from Git (without removing them locally) and archive them to a configured sibling specs repository.
  * Added an interactive `--sync` workflow to archive completed brainstorm documents to `brainstorm/attic/` and refresh the overview.
* **Documentation**
  * Updated README and command/help text for stealth mode (`enable`/`archive`/`is-enabled`) and finish-time archiving behavior; revised `spex-finish` option descriptions.
* **Bug Fixes**
  * Improved brainstorm sync behavior and clarified Codex external-tool detection formatting.
* **Tests**
  * Added coverage for stealth-mode enablement/exclude behavior and archiving (including skip/idempotency/edge cases).

### CODE REVIEW

File: spex/extensions/spex/commands/speckit.spex.finish.md [hunk 1]
```diff
@@ -169,73 +169,41 @@ Assisted-By: 🤖 Claude Code"
...
 ```bash
 DETACH_ENABLED=false
-DETACH_PR_BRANCH=""
 if [ -d ".specify/extensions/spex-detach" ]; then
   DETACH_ENABLED=true
 fi
 ```
 
-If `DETACH_ENABLED` is true, run the detach flow. The ordering matters: archive copies artifacts to the sibling repo first (without deleting sources), then detach creates the clean PR branch, and only after the PR branch is verified clean do we delete the archived source directories. This ensures FR-004 compliance (source deletion occurs after PR branch creation and verification).
+If `DETACH_ENABLED` is true, archive spec artifacts to the sibling specs repo (if configured). With stealth mode, spec files are excluded from git via `.git/info/exclude`, so no clean PR branch is needed.
```
[the remark this answers is not in the document]
Comment (rhuss) [reply spex/extensions/spex/commands/speckit.spex.finish.md:183]: In stealth mode, spec files are excluded via `.git/info/exclude` before they are ever committed. The `git add -A` at line 164 skips excluded paths, so spec files never enter the index. The `enable` subcommand warns if spec files are already tracked (FR-005 in the spec), covering the migration scenario from the old detach approach. No code change needed.

File: spex/extensions/spex/commands/speckit.spex.finish.md [hunk 2]
```diff
@@ -169,73 +169,41 @@ Assisted-By: 🤖 Claude Code"
...
-         ARCHIVE_SUCCEEDED=true
          echo "Archive: specs copied to sibling repo"
        else
-         echo "WARNING: Archive failed, continuing with detach"
+         echo "WARNING: Archive failed, continuing with finish"
        fi
      else
        echo "WARNING: archive.path not configured or not reachable, skipping archive"
      fi
    fi
```
[the remark this answers is not in the document]
Comment (rhuss) [reply spex/extensions/spex/commands/speckit.spex.finish.md:206]: Good catch. With stealth mode, spec files are excluded from git and the sibling repo archive is the only version control mechanism. Silently continuing after archive failure would leave specs unversioned. Changed to a hard stop with a clear error message, with `--skip-archive` as the explicit override.

Applied in 5aed97c.
